flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/CEP.java

```diff
@@ -52,7 +52,48 @@
 			DataStream<T> input,
 			Pattern<T, ?> pattern,
 			EventComparator<T> comparator) {
-		final PatternStream<T> stream = new PatternStream<>(input, pattern);
-		return stream.withComparator(comparator);
+		return new PatternStream<>(input, pattern).withComparator(comparator);
+	}
+
+	/**
+	 * Creates a {@link PatternStream} from an input data stream and a pattern.
+	 *
+	 * @param input DataStream containing the input events
+	 * @param pattern Pattern specification which shall be detected
+	 * @param isProcessingTime Time behaviour to specify processing time or event time
+	 * @param <T> Type of the input events
+	 * @return Resulting pattern stream
+	 */
+	public static <T> PatternStream<T> pattern(
+			DataStream<T> input,
+			Pattern<T, ?> pattern,
+			boolean isProcessingTime) {
```

Comment:
I think we don't need to expose an additional parameter because users can use `inProcessingTime()` and `inEventTime()` right?

Reply:
It's better to just have one way of doing things. Keeps the API simpler. What do you think?

Reply:
@aljoscha I thought that this could provide this interface to directly create PatternStream in processing time with parameter, and this is user-friendly interface. Your point that keeping the API simpler is right, but for users, IMO, it's better to specify the time behaviour by parameter, not extrally calling `inProcessingTime()` and `inEventTime()`.

Reply:
I still think having this as a parameter would not be very user friendly, seeing a call like `pattern(in, foo, false)` makes it very hard to know what the `false` means. If you see, however, `pattern(in, foo).inProcessingTime()` it's immediately obvious what's going on.

Reply:
@aljoscha I would like to follow up the points you mentioned later to update the methods.